### PR TITLE
SDK version update

### PR DESCRIPTION
**Pull request by drwill-ms**
## Microsoft.Azure.Devices v1.20.0

- Enables new security functionality to specify how hub should authenticate when working with a storage account when running import/export devices jobs. Initially, this feature has limited availability, starting with 3 public cloud Azure regions. Since it is not yet widely available, this functionality must be turned on. See below (1) for details.
  - New enum `StorageAuthenticationType` to specify key-based or identity-based authentication.
  - New property `JobProperties.StorageAuthenticationType` for use in import/export devices jobs.
  - Added two overloaded methods: `RegistryManager.ImportDevicesAsync` and `RegistryManager.ExportDevicesAsync`. The overloads take a `JobProperties` instance so the user can pass all required and any optional parameters.
  - Two factory methods to make it easier to call `ImportDevicesAsync` and `ExportDevicesAsync`, making it clear which properties are required and which are optional. They are `JobProperties.CreateForInportJob` and `JobProperties.CreateForExportJob` respectively.
  - Removed two unreferenced classes: `ImportDevicesRequest` and `ExportDevicesRequest`.
- `IotHubClientWebSocket.ConnectAsync` method now respects the `SslProtocol` value specified in `TlsVersions.Instance.Preferred`.
- Dropped support for .NET Standard 1.3, and added .NET Framework 4.7.2.
- Updated reference to `Microsoft.Azure.Devices.Shared` nuget.

> (1) To enable this functionality, this version of the SDK will require setting an environment variable of `EnableStorageIdentity` to `1`. Once this feature is widely available, this requirement will be removed.

## Microsoft.Azure.Devices.Shared v1.19.0

- Dropped support for .NET Standard 1.3, and added .NET Framework 4.7.2.

## Microsoft.Azure.Devices.Client v1.24.0

- Dropped support for .NET Standard 1.3, and added .NET Framework 4.7.2.
- Updated reference to `Microsoft.Azure.Devices.Shared` nuget.

## Microsoft.Azure.Devices.Provisioning.Client v1.5.1

- Updated reference to `Microsoft.Azure.Devices.Shared` nuget.

## Microsoft.Azure.Devices.Provisioning.Service v1.6.1

- Updated reference to `Microsoft.Azure.Devices.Shared` nuget.

## Microsoft.Azure.Devices.Provisioning.Transport.Amqp v1.2.1

- Updated reference to `Microsoft.Azure.Devices.Shared` nuget.

## Microsoft.Azure.Devices.Provisioning.Transport.Mqtt v1.2.1

- Updated reference to `Microsoft.Azure.Devices.Shared` nuget.

## Microsoft.Azure.Devices.Provisioning.Transport.Http v1.2.1

- Updated reference to `Microsoft.Azure.Devices.Shared` nuget.

## Microsoft.Azure.Devices.Provisioning.Security.Tpm v1.2.1

- Updated reference to `Microsoft.Azure.Devices.Shared` nuget.

### CODE REVIEW

File: versions.csv [hunk 1]
```diff
@@ -1,10 +1,10 @@
 AssemblyPath, Version
 iothub\device\src\Microsoft.Azure.Devices.Client.csproj, 1.23.2
```
Comment (abhipsaMisra) [inline versions.csv]: patch version bump here for device client

Comment (drwill-ms) [reply versions.csv]: thanks

File: versions.csv [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-iothub\device\src\Microsoft.Azure.Devices.Client.csproj, 1.23.2
-iothub\service\src\Microsoft.Azure.Devices.csproj, 1.19.0
-shared\src\Microsoft.Azure.Devices.Shared.csproj, 1.18.0
-provisioning\device\src\Microsoft.Azure.Devices.Provisioning.Client.csproj, 1.5.0
-provisioning\transport\amqp\src\Microsoft.Azure.Devices.Provisioning.Transport.Amqp.csproj, 1.2.0
-provisioning\transport\http\src\Microsoft.Azure.Devices.Provisioning.Transport.Http.csproj, 1.2.0
-provisioning\transport\mqtt\src\Microsoft.Azure.Devices.Provisioning.Transport.Mqtt.csproj, 1.2.0
-security\tpm\src\Microsoft.Azure.Devices.Provisioning.Security.Tpm.csproj, 1.2.0
-provisioning\service\src\Microsoft.Azure.Devices.Provisioning.Service.csproj, 1.6.0
+iothub\device\src\Microsoft.Azure.Devices.Client.csproj, 1.23.3
```
Comment (abhipsaMisra) [inline versions.csv]: minor version bump - needs to be 1.24.0

**Pull request merged**